build/yamls/antrea-aks.yml

```diff
@@ -571,6 +571,14 @@ rules:
   verbs:
   - get
   - list
+- apiGroups:
+  - metrics.antrea.tanzu.vmware.com
+  resources:
+  - networkpolicymetrics
```

Comment:
Question - so it covers both K8s and Antrea NP?

Reply:
But how we are going to fix it? I prefer not to use UID as Name in user/client facing API.

Reply:
But on the hand, 3 strings - type, namespace, name - can lead to much more traffic, esp. for client/collector polls all stats...

Reply:
networkpolicymetrics will have metrics of K8s networkpolicies only, I think we don't want to mix different resources in one API as we need to distinguish from ClusterNetworkPolicy anyway.
There will be another API antreanetworkpolicymetrics for antrea namespaced networkpolicies like below. Feel free to suggest another name.
```
// get K8s NP metrics
kubectl get networkpolicymetrics
// get CNP metrics
kubectl get clusternetworkpolicymetrics
// get ANP metrics
kubectl get antreanetworkpolicymetrics
```
For the length of the key, do you mean we don't set its category and use UID to figure it out? If yes, it requires that all metrics are stored in same store, and will not be efficient to execute the query. For example, the query of listing all metrics of K8s networkpolicies of a specified namespace will need two indices, one for category and one for namespace, but the cache.Indexer doesn't support it yet.

How about using 3 slices to report the stats to controller?
```
// The TrafficStats of K8s NetworkPolicies.
NetworkPolicies []NetworkPolicyStats
// The TrafficStats of Antrea ClusterNetworkPolicies.
ClusterNetworkPolicies []NetworkPolicyStats
// The TrafficStats of Antrea NetworkPolicies.
AntreaNetworkPolicies []NetworkPolicyStats
```

Reply:
Three types sound good to me.

For the UID question, even we must have a separate type (guess protobuf will optimize the message field IDs?), should we try saving some bytes by using UID, instead of Namespace + name? I know even UID is not short as string though.
Just feel stats polling might generate lots of traffic. Maybe you already have some estimate.

Reply:
We could use UID, I think it's more accurate for an asynchronous stats collection. I don't worry too much about the length of the stats because normally there won't be too many policies applied to each Node, say 10 policies per Pod, 100 Pods per Node, the namespace+name or UID takes 40 chars, then each stats summary is about 10*100*(40+8*3)=64KB (protobuf might reduce the size of ints further), while a single Endpointslice with 100 Endpoints can get ~10KB and it should be updated more frequently than the stats polling. And we could increase the polling period if performance issue is seen.

Reply:
Ok. 64KB sounds not bad. Maybe better to dump the message to check the real size.

Reply:
@jianjuns I tested Marshal 1 NodeStatsSummary that contains 1000 policies with only UID set:
```
func TestNodeStatsSummarySize(t *testing.T) {
	summary := &cpv1alpha1.NodeStatsSummary{}
	for i := 0; i < 1000; i++ {
		stats := cpv1alpha1.NetworkPolicyStats{
			NetworkPolicy: cpv1alpha1.NetworkPolicyReference{
				UID: "561e923c-a9c6-43b6-80d8-fda1427e4b98",
			},
			TrafficStats:  metricsv1alpha1.TrafficStats{
				Sessions: 1000,
				Packets: 100000,
				Bytes: 100000000,
			},
		}
		summary.NetworkPolicies = append(summary.NetworkPolicies, stats)
	}
	bytes, _ := summary.Marshal()
	t.Logf("Bytes: %v", len(bytes))
}
```
The result is `62020`.

If with the longest type (AntreaClusterNetworkPolicy) set in NetworkPolicyReference, the result is `88020`.

Reply:
Ok. Thanks for the update.